src/scanspec/specs.py

```diff
@@ -667,3 +667,31 @@ def step(spec: Spec[Axis], duration: float, num: int = 1) -> Spec[Axis]:
         spec = step(Line("x", 1, 2, 3), 0.1)
     """
     return spec * Static.duration(duration, num)
+
+
+def duration(frames: List[Frames]) -> Optional[float]:
```

Comment:
Actually, should this be called something like `get_constant_duration`?

Reply:
I had renamed the function, but I've put it back to get_constant_duration.